### PR TITLE
WIP: #378

### PR DESCRIPTION
**Pull request by jeremyabbott**
@haf would you mind taking a look at this? This is my first pass based on our conversation.

Thanks in advance!

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+    #{p.proj_path_base}/bin/#{Configuration} ==> lib"
+    template = open("paket.template", "w")
+    template.write(m)
+    template.close
+    system "tools/paket.exe pack output build/pkg"
+    # p.declared_packages.each do |d|
+    #     puts d
+    # end
+   end
+   File.delete "paket.template"
```
Comment (haf) [inline Rakefile]: being - ensure should be used, in case writing file fails

File: Rakefile [hunk 2]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+authors Ademar Gonzalez, Henrik Feldt
+description #{suave_description}
+language en-GB
+copyright Ademar Gonzalez, Henrik Feldt
+licenseUrl https://github.com/SuaveIO/Suave/blob/master/COPYING
+projectUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+iconUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+files
+    #{p.proj_path_base}/bin/#{Configuration} ==> lib"
+    template = open("paket.template", "w")
```
Comment (haf) [inline Rakefile]: Prefer qualified `File.open do |template| ... end`

File: Rakefile [hunk 3]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+    template.write(m)
```
Comment (haf) [inline Rakefile]: No brackets needed

File: Rakefile [hunk 4]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+version #{ENV['NUGET_VERSION']}
+authors Ademar Gonzalez, Henrik Feldt
+description #{suave_description}
+language en-GB
+copyright Ademar Gonzalez, Henrik Feldt
+licenseUrl https://github.com/SuaveIO/Suave/blob/master/COPYING
+projectUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+iconUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+files
+    #{p.proj_path_base}/bin/#{Configuration} ==> lib"
```
Comment (haf) [inline Rakefile]: Nice, does it work for CoreCLR? I don't mind if it doesn't right now as we're not publishing nugets from such a compilation.

File: Rakefile [hunk 5]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
   end
 end
 
+task :paket => [:versioning, 'build/pkg'] do
+   puts "starting"
+   files = FileList['src/**/*.fsproj'].exclude(/Tests/)
```
Comment (haf) [inline Rakefile]: Good

File: Rakefile [hunk 6]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
   end
 end
 
+task :paket => [:versioning, 'build/pkg'] do
```
Comment (haf) [inline Rakefile]: Good

File: Rakefile [hunk 7]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+   puts "starting"
+   files = FileList['src/**/*.fsproj'].exclude(/Tests/)
+   files.each do |f|
+    p = Albacore::Project.new f
+    m = "type file
```
Comment (haf) [inline Rakefile]: If you want to use quotes in the string, you can also start the string with `%{` and end with `}`, FYI

File: Rakefile [hunk 8]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+language en-GB
+copyright Ademar Gonzalez, Henrik Feldt
+licenseUrl https://github.com/SuaveIO/Suave/blob/master/COPYING
+projectUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+iconUrl https://raw.githubusercontent.com/SuaveIO/resources/master/images/head_trans.png
+files
+    #{p.proj_path_base}/bin/#{Configuration} ==> lib"
+    template = open("paket.template", "w")
+    template.write(m)
+    template.close
```
Comment (haf) [inline Rakefile]: will be closed by `end`

File: Rakefile [hunk 9]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+    system "tools/paket.exe pack output build/pkg"
```
Comment (haf) [inline Rakefile]: Looking good. Alternative formulation that skips string parsing `system "tools/paket.exe", %w|pack output build/pkg|, clr_command: true`, which also will work cross-platform (unix + windows)

File: Rakefile [hunk 10]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+    # p.declared_packages.each do |d|
```
Comment (haf) [inline Rakefile]: Remove comments?

File: Rakefile [hunk 11]
```diff
@@ -136,6 +137,35 @@ nugets_pack :create_nuget_quick => [:versioning, 'build/pkg'] do |p|
...
+    #     puts d
+    # end
+   end
+   File.delete "paket.template"
+   puts "fin"
```
Comment (haf) [inline Rakefile]: Awesome, Finnish for the WIN!

**Comment by haf** [comment]
@jeremyabbott Sweet :+1: !

Put some comments in there.

**Comment by jeremyabbott** [comment]
@haf thanks for looking at it. It's definitely not done. I just wanted to make sure I was on the right path, and working towards what you were expecting.

**Comment by haf** [comment]
Yes, this is great.

File: Rakefile (hunk 4 above)
Comment (jeremyabbott) [inline Rakefile]: I actually haven't tried.

File: Rakefile (hunk 6 above)
Comment (jeremyabbott) [inline Rakefile]: This will replace `:create_nuget_quick` when I'm finished.

**Comment by jeremyabbott** [comment]
@haf I got everything working (dependencies, files, code review feedback). Would you mind letting me know what you think?

**Comment by haf** [comment]
I think the code looks OK. Only two things:

 - use 2-space indentation
 - can we move the new .rb file to `./tools`?

Otherwise +1!

**Comment by jeremyabbott** [comment]
@haf updates made. Changes squashed.

**Comment by haf** [comment]
@jeremyabbott Thank you

**Pull request merged**